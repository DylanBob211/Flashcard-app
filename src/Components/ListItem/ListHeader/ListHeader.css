
.listItem_header_container {
  border-radius: 10px 10px 0 0;
  display: flex;
  align-items: center;
  background-color: var(--primary-color)
}

.listItem_title {
  flex-grow: 4;
  text-align: center;;
  color: #fff;

}

.listItem_iconbox{
  flex-grow: 1;
  align-items: center;
}

.listItem_icon--trashbin {
  height: 20px;
  fill: #fff;
  transition: .3s fill;
}

.listItem_icon--play {
  height: 20px;
  margin-left: .2em;
  fill: #fff;
  transition: .3s fill;
}

.listItem_icon--trashbin:hover {
  fill: var(--first-shade-clashing);
  cursor: pointer;
}

.listItem_icon--play:hover {
  fill: var(--second-shade-secondary);
  cursor: pointer;
}
