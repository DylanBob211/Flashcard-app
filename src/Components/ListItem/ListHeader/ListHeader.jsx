import React from 'react';
import PropTypes from 'prop-types';
import TrashBinIcon from '../../Icons/TrashBinIcon';
import PlayIcon from '../../Icons/PlayIcon';
import './ListHeader.css';


const ListHeader = ({ listName, deleteList, openExerciseWindow }) => (
  <div className="listItem_header_container">
    <h2 className="listItem_title">{ listName }</h2>
    <div className="listItem_iconbox">
      <TrashBinIcon
        data-test="deleteListIconButton"
        disabled={!deleteList}
        onClick={() => deleteList(listName)}
        className="listItem_icon--trashbin"
      />
      <PlayIcon
        disabled={!openExerciseWindow}
        className="listItem_icon--play"
        onClick={openExerciseWindow}
      />
    </div>
  </div>
);

ListHeader.propTypes = {
  listName: PropTypes.string,
  deleteList: PropTypes.func,
  openExerciseWindow: PropTypes.func,
};

ListHeader.defaultProps = {
  listName: '',
  deleteList: null,
  openExerciseWindow: null,
};

export default ListHeader;
