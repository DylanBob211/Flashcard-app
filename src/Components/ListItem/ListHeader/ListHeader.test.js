import React from 'react';
import { mount, shallow } from 'enzyme';
import ListHeader from './ListHeader';

describe('ListHeader', () => {
  let mountedListHeader;

  const listHeader = (props, isShallow = false) => {
    const jsx = (
      <ListHeader
        {...props}
      />
    );
    if (isShallow) return shallow(jsx);
    return mount(jsx);
  };

  const props = {
    listName: '',
    deleteList: jest.fn(),
    openExerciseWindow: jest.fn(),
  };

  beforeEach(() => {
    mountedListHeader = undefined;
    jest.clearAllMocks();
  });

  it('matches to snapshot', () => {
    expect(listHeader(props, true)).toMatchSnapshot();
  });

  it('always renders a div container', () => {
    const divContainer = listHeader(props).find('.listItem_header_container');
    expect(divContainer.exists()).toEqual(true);
  });

  describe('when `listName` is passed', () => {

  });
});
