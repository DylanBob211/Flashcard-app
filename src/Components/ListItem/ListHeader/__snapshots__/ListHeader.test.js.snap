// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeader matches to snapshot 1`] = `
<div
  className="listItem_header_container"
>
  <h2
    className="listItem_title"
  />
  <div
    className="listItem_iconbox"
  >
    <TrashBinIcon
      className="listItem_icon--trashbin"
      data-test="deleteListIconButton"
      disabled={true}
      onClick={[Function]}
    />
    <PlayIcon
      className="listItem_icon--play"
      disabled={true}
      onClick={null}
    />
  </div>
</div>
`;
